### PR TITLE
Add ability to override github app callback url

### PR DESCRIPTION
**Pull request by nicolasarnold12321**
By default, the github app provider redirects back to the base url.
This change allows a user to override the GitHub App url using the config field `callbackUrl`.

It will look like this in the configuration:
```
auth:
  providers:
    github:
      development:
        clientId: SUPER ID
        clientSecret: SUPER SECRET
        callbackUrl: https://some-url.com/callback
```

Signed-off-by: Nicolas Arnold <nic@roadie.io>

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Rugvip — COMMENTED** [review]
Thank you!

May I ask why you need this? 😁

**Comment by nicolasarnold12321** [comment]
Hi @Rugvip . We have a custom GitHub app that serves requests for many backstage instances. GitHub apps only allows one redirect url for the whole application. Due to this, we require an intermediate service that verifies the installation request and redirects it back to the correct domain.

**Comment by Rugvip** [comment]
@nicolasarnold12321 ah alright. I think we could just call it `callbackUrl` though, and best add it to the GitHub auth provider docs with a disclaimer that you should only use this if your Backstage instance is not the initial receiver of the auth callback

**Comment by nicolasarnold12321** [comment]
@Rugvip Perfect. Thank you. I'll make those changes now.

File: docs/auth/github/provider.md [hunk 1]
```diff
@@ -50,6 +52,10 @@ The GitHub provider is a structure with three configuration keys:
 - `clientSecret`: The client secret tied to the generated client ID.
 - `enterpriseInstanceUrl` (optional): The base URL for a GitHub Enterprise
   instance, e.g. `https://ghe.<company>.com`. Only needed for GitHub Enterprise.
+- `callbackUrl` (optional): The callback url that GitHub will use when
```
Comment (Rugvip) [inline docs/auth/github/provider.md:55]: Tbh I think this is such a niche use case that we should only document it here avoid skip the two examples above and in the other doc. It's unfortunate if people end up struggling with configuring the callback URL when there's no need

**Review by Rugvip — COMMENTED** [review]
Thanks! 👍 

Need the DCO as well, see the details on the failing check to see how to get that sorted

File: docs/auth/github/provider.md (hunk 1 above)
Comment (nicolasarnold12321) [reply docs/auth/github/provider.md:55]: Ok sounds good.

Comment (nicolasarnold12321) [reply docs/auth/github/provider.md:55]: Done!

**Comment by nicolasarnold12321** [comment]
I don't have merge permissions @Rugvip . So if you're happy with the changes, would you mind merging for me?

File: docs/auth/github/provider.md [hunk 2]
```diff
@@ -41,6 +41,8 @@ auth:
         clientSecret: ${AUTH_GITHUB_CLIENT_SECRET}
         ## uncomment if using GitHub Enterprise
         # enterpriseInstanceUrl: ${AUTH_GITHUB_ENTERPRISE_INSTANCE_URL}
+        ## uncomment if intermediate service is used when using github apps
+        # callbackUrl: ${AUTH_GITHUB_CALLBACK_URL}
```
Comment (Rugvip) [inline docs/auth/github/provider.md:45]: This one too 😅

Comment (nicolasarnold12321) [reply docs/auth/github/provider.md:45]: Ok! I'll remove this then so. Thanks

**Review by Rugvip — APPROVED** [review]
:shipit: !

**Comment by nicolasarnold12321** [comment]
Thank you @Rugvip

**Pull request merged**